india_compliance/gst_india/client_scripts/e_waybill_applicability.js

```diff
@@ -0,0 +1,87 @@
+class EwaybillApplicability {
+    constructor(frm) {
+        this.frm = frm;
+    }
+    is_e_waybill_applicable() {
+        if (
+            // means company is Indian and not Unregistered
+            !this.frm.doc.company_gstin ||
+            !gst_settings.enable_e_waybill ||
+            this.frm.doc.is_opening === "Yes"
+        )
+            return false;
+
+        // at least one item is not a service
+        for (const item of this.frm.doc.items) {
+            if (
+                item.gst_hsn_code &&
+                !item.gst_hsn_code.startsWith("99") &&
+                item.qty !== 0
+            )
+                return true;
+        }
+
+        return false;
+    }
+    is_e_waybill_generatable_using_api() {
+        return this.is_e_waybill_applicable && india_compliance.is_api_enabled();
+    }
+
+    auto_generate_e_waybill() {
+        return false;
+    }
+}
+
+class SalesInvoiceEwaybill extends EwaybillApplicability {
+    is_e_waybill_generatable_using_api() {
+        return (
+            super.is_e_waybill_generatable_using_api() &&
+            this.frm.doc.company_gstin !== this.frm.doc.customer_gstin &&
+            this.frm.doc.customer_address
```

Comment:
With this, the Generate button will not be visible. Instead, it should throw while trying to generate. It would confuse users if button is not visible.